### DISCUSSION TITLE
Restricting SSH to specific IPs instead of anywhere

### DISCUSSION THREAD

**Discussion by McLoggin** [Security-Onion-Solutions/securityonion#3886] [Unsupported Versions]
In Security Onion 16.04, we are able to restrict the IPs that could access our nodes via SSH with ufw. In Security Onion 2, we noticed that for every node type there is a hostgroup called anywhere and the portgroup ssh is allowed to access all of the node types. We understand that ssh is required for initial setup for the nodes, so we would have to add another hostgroup and add all of the sensor's IPs to that hostgroup to access our searchmanager. 

The issue that we're running into is that if we wanted to take out the anywhere hostgroup which is only called for ssh, we would have to take it out of /opt/so/saltstack/default/salt/firewall/assigned_hostgroups.map.yaml and it would be overwritten when we patched. I've tried adding an INPUT chain to /opt/so/saltstack/local/salt/firewall/assigned_hostgroups.local.map.yaml to the anywhere hostgroup and leaving it blank to see if it would overwrite the rule from default, which didn't work. We added our own hostgroup that restricted ssh to a specific group of IPs that only the admins would have access from and verified that it was being passed via salt. Its still passing the rule for the anywhere hostgroup from default file.

Another thing we tried was to copy the default assigned_hostgroups.map.yaml to local assigned_hostgroups.local.map.yaml and take away the anywhere hostgroup and add our own hostgroup with only the admin IPs. Unfortunately, it still is passing the iptables rule from the default file. Has anyone tried restricting their ssh access to a specific set of IPs or know a way where we can just edit the local file to keep our configurations when we patch?

**Comment by TOoSmOotH** [comment]
I have created #3915 to add this as a feature.
